packages/test-sync/src/androidTest/kotlin/io/realm/kotlin/test/shared/SyncedRealmTests.kt

```diff
@@ -969,6 +1044,76 @@ class SyncedRealmTests {
         }
     }
 
+    // Test for https://github.com/realm/realm-kotlin/issues/1068
+    // Note, this test is not 100% sure to surface the bug, but manual testing has shown that it
+    // works well enough. Also, even if it doesn't surface the bug, it will not the fail the test.
+    @Test
+    fun accessSessionAfterRemoteChange() = runBlocking {
+        val flexApp = TestApp(
+            appName = io.realm.kotlin.test.mongodb.TEST_APP_FLEX,
+            builder = {
+                it.syncRootDirectory(PlatformUtils.createTempDir("flx-sync-"))
+            }
+        )
+        val section = Random.nextInt()
+        val (email1, password1) = randomEmail() to "password1234"
+        val (email2, password2) = randomEmail() to "password1234"
+        val user1 = flexApp.createUserAndLogIn(email1, password1)
+        val user2 = flexApp.createUserAndLogIn(email2, password2)
+        val syncConfig1 = createFlexibleSyncConfig(
+            user = user1,
+            name = "sync1.realm",
+            initialSubscriptions = { realm: Realm ->
+                realm.query<FlexParentObject>("section = $0", section).subscribe()
+            }
+        )
+        val syncConfig2 = createFlexibleSyncConfig(
+            user = user2,
+            name = "sync2.realm",
+            initialSubscriptions = { realm: Realm ->
+                realm.query<FlexParentObject>("section = $0", section).subscribe()
+            }
+        )
+        val realm1 = Realm.open(syncConfig1)
+
+        Realm.open(syncConfig2).use { realm2 ->
+            realm2.write {
+                copyToRealm(FlexParentObject(section))
+            }
+            realm2.syncSession.uploadAllLocalChanges()
+        }
+
+        // Reading the object means we received it from the other Realm
+        withTimeout(30.seconds) {
+            val obj: FlexParentObject = realm1.query<FlexParentObject>("section = $0", section).asFlow()
+                .map { it.list }
+                .filter { it.isNotEmpty() }
+                .first().first()
+            assertEquals(section, obj.section)
+
+            // 1. Local write to work around https://github.com/realm/realm-kotlin/issues/1070
+            realm1.write { }
+
+            // 2. Trigger GC. This will GC the RealmReference JVM object, making the native reference
+            //    eligible for closing.
+            PlatformUtils.triggerGC()
+
+            // 3. On the next update of Realm, we run through the weak list of all previous
+            //    RealmReferences and close all native pointers with their JVM object GC'ed.
+            //    This should now include the object created in step 1.
+            realm1.write { }
+        }
+
+        // 4. With the original native dbPointer now being closed, accessing the syncSession for
+        //    the first time should still work.
+        try {
+            realm1.syncSession.pause()
```

Comment:
If an error happens here it will get swallowed by the `finally` and we won't know whether the session is still accessible. I would check the session state has actually been updated after calling `pause`.

Reply:
Hmm, `finally` doesn't swallow the exception, it should still propagate and fail the test, but you are right. Checking the state would improve the test.

Reply:
Yes, I didn't mean an exception gets swallowed but rather that the state itself doesn't get updated for some reason. Poor wording 🙊